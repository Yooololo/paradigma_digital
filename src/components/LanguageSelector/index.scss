.LanguageSelector {
  .flag {
    width: $margin + $padding;
    margin: 0 $margin;
    cursor: pointer;
    &.light {
      border: 2px solid $black;
    }
    &.dark {
      border: 2px solid $white;
    }
  }
}
