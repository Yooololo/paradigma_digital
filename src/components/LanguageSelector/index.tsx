import { useContext } from "react";
import { ThemeColorContext } from "../../context/ThemeColorWrapper";
import { LanguageContext } from "../../context/LanguageWrapper";

export default function LanguageSelector() {
  const { themeColor } = useContext(ThemeColorContext);
  const { selectLanguage } = useContext(LanguageContext);

  return (
    <div className={`LanguageSelector ${themeColor}`}>
      <img
        className={`flag ${themeColor}`}
        onClick={() => selectLanguage("en")}
        alt="en"
        src="http://purecatamphetamine.github.io/country-flag-icons/3x2/GB.svg"
      />
      <img
        className={`flag ${themeColor}`}
        onClick={() => selectLanguage("es")}
        alt="es"
        src="http://purecatamphetamine.github.io/country-flag-icons/3x2/ES.svg"
      />
      <img
        className={`flag ${themeColor}`}
        onClick={() => selectLanguage("zh")}
        alt="zh"
        src="http://purecatamphetamine.github.io/country-flag-icons/3x2/CN.svg"
      />
    </div>
  );
}
