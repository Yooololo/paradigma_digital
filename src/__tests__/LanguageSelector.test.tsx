import LanguageSelector from "../components/LanguageSelector";
import { cleanup, render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";

afterEach(cleanup);

test("LanguageSelector renders correctly", () => {
  const { container } = render(
    <IntlProvider locale="en-US" messages={English}>
      <LanguageSelector />
    </IntlProvider>
  );
  expect(container).toMatchSnapshot();
  expect(container.querySelectorAll(".flag")).toHaveLength(3);
});
