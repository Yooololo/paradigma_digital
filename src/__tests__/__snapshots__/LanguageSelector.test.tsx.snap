// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector renders correctly 1`] = `
<div>
  <div
    class="LanguageSelector dark"
  >
    <img
      alt="en"
      class="flag dark"
      src="http://purecatamphetamine.github.io/country-flag-icons/3x2/GB.svg"
    />
    <img
      alt="es"
      class="flag dark"
      src="http://purecatamphetamine.github.io/country-flag-icons/3x2/ES.svg"
    />
    <img
      alt="zh"
      class="flag dark"
      src="http://purecatamphetamine.github.io/country-flag-icons/3x2/CN.svg"
    />
  </div>
</div>
`;
